### DISCUSSION TITLE
Tables not showing after fetch from database

### DISCUSSION THREAD

**Discussion by smokey58au** [Budibase/budibase#11836] [Help] [answered]
I'm only new to BB, so it might be something I'm doing wrong, but when I try and fetch the tables from my MariaDB database they aren't displaying in the Sources panel on the left (under my MYSQL source), so I can't use them for anything.  When I click Fetch Tables it comes up (correctly) with a list of the valid tables to choose from, and when I select the tables (some or all) it says "Tables Fetched Successfully" but they don't appear under Sources or anywhere else.  I believe I'm connecting correctly because the panel at the top of the page for my MySQL source says "Connected" with a green tick, and the list of tables appears when I start the Fetch process.  I'm running MariaDB 10.8 on a Ubuntu 22.04 VM, and BB is in a docker container on the same VM.  Is anyone able to help with what might be wrong?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @smokey58au,

If you can see the tables in the modal were you have the option to select them. Then yes you have setup the connection correctly.

Do your tables have `primary keys` set? Budibase can't sync in tables which don't have a `primary key` field. I created an issue earlier today, to add an error message to explain this to the user. This issue can be found here; https://github.com/Budibase/budibase/issues/11843

I hope this helps.

**Reply by smokey58au** [reply]
Thanks @ConorWebb96, this was causing the problem.  All good now, I appreciate your help.
